clang/lib/3C/ProgramInfo.cpp

```diff
@@ -1036,15 +1035,26 @@ ProgramInfo::getTypeParamBindings(CallExpr *CE, ASTContext *C) const {
   return TypeParamBindings.at(PSL);
 }
 
-std::pair<CVarSet, bool> ProgramInfo::lookupTypedef(PersistentSourceLoc PSL) {
-  return typedefVars[PSL];
+CVarOption ProgramInfo::lookupTypedef(PersistentSourceLoc PSL) {
+  return TypedefVars[PSL];
 }
 
 bool ProgramInfo::seenTypedef(PersistentSourceLoc PSL) {
-  return typedefVars.count(PSL) != 0;
+  return TypedefVars.count(PSL) != 0;
 }
 
-void ProgramInfo::addTypedef(PersistentSourceLoc PSL, bool ShouldCheck) {
-  CVarSet empty;
-  typedefVars[PSL] = make_pair(empty, ShouldCheck);
+void ProgramInfo::addTypedef(PersistentSourceLoc PSL, bool CanRewriteDef,
+                             TypedefDecl* TD, ASTContext &C) {
+  auto Name = "typedef__" + TD->getNameAsString();
```

Comment:
Was there a reason for the `typedef__` prefix?  Removing it made it easier for me to implement #441.

(If you like, you could go ahead and cherry-pick all 3 lines of #441 into this PR and add a regression test for the bug it's fixing, but I'm also happy to finish #441 separately later.)

Reply:
The only reason is I thought it might make things easier to debug constraints if they were named usefully.

Reply:
I reason that `PointerVariableConstraint::mkString` works under the assumption that the `PointerVariableConstraint::Name` is the original name of the program element, and we already use it that way for normal variables.  So assuming we want to fix #441 eventually, it would be easiest to go ahead and have typedefs follow that same assumption and add a separate name mangling mechanism for debug output if we want.

Reply:
We can implement whatever name mangling we like on atom names I believe. Those are only ever used for debug output.